### PR TITLE
Cleanup noise interfaces and implementations

### PR DESCRIPTION
**Pull request by msteiger**
This PR updates the structure of noise classes.

* 2D and 3D Noise are merged in one interface. A default implementation for 2D noise delegates to the 3D version by setting z = constant.
* Two methods for integer noise are added, which are useful for per-block noise. By default, ints are mapped to float values.
* Added `WhiteNoise` (infinite resolution) and `DiscreteWhiteNoise` (integer resolution) based on a hashing function that is superior to `FastNoise` and `NoiseTable`.
* **Breaking change** Unified ranges of return values. BrownianNoise returned -scale..+scale before,  FastNoise was [0..1] before. Now all are in [-1..1].
* Added more tests to verify generated value ranges, etc.
* Old classes/methods were left as-is, but annotated with `@Deprecated`

This PR requires #1572 (JUnit 4.11 to be precise)

I've added some documentation on the wiki based on the visual comparison table, which shows different implementations in 10x magnification.

noise_comparison

### CODE REVIEW

**Comment by Cervator** [comment]
Dependency is in place and I saw the image earlier, very cool.

No objections here, haven't tested yet though. Pinging @immortius and some world gen peeps like @Josharias @MarcinSc @skaldarnar for quick review

File: engine/src/main/java/org/terasology/utilities/procedural/WhiteNoise.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @Override
+    public float noise(int x, int y, int z) {
+        return noise(intNoise(intNoise(x) + y) + z);
+    }
+
+    /**
+     * @param x any float value
+     * @return a random, but deterministic float value in [-1..1]
+     */
+    public float noise(float x) {
```
Comment (Josharias) [inline engine/src/main/java/org/terasology/utilities/procedural/WhiteNoise.java]: Do we care that the 1 dimension version of noise is not in the main Noise interface/abstract class?

File: modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinBaseSurfaceProvider.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public void process(GeneratingRegion region) {
         float[] noise = surfaceNoise.noise(processRegion);
 
         for (int i = 0; i < noise.length; ++i) {
-            noise[i] = seaLevel + seaLevel * ((noise[i] + 1f) / 2f);
+            noise[i] = seaLevel + seaLevel * ((noise[i] * 2.11f + 1f) / 2f);
```
Comment (Josharias) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinBaseSurfaceProvider.java]: What does this magic number do?  Just make everything more amazingly deep/high?

File: modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinHillsAndMountainsProvider.java [hunk 3]
```diff
@@ -69,8 +69,8 @@ public void process(GeneratingRegion region) {
             float tempHumid = temp * humidityData.get(pos);
             Vector2f distanceToMountainBiome = new Vector2f(temp - 0.25f, tempHumid - 0.35f);
             float mIntens = TeraMath.clamp(1.0f - distanceToMountainBiome.length() * 3.0f);
-            float densityMountains = Math.max(mountainData[i], 0) * mIntens * configuration.mountainAmplitude;
-            float densityHills = Math.max(hillData[i] - 0.1f, 0) * (1.0f - mIntens) * configuration.hillAmplitude;
+            float densityMountains = Math.max(mountainData[i] * 2.12f, 0) * mIntens * configuration.mountainAmplitude;
```
Comment (Josharias) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinHillsAndMountainsProvider.java]: 2.11f was not good enough?  had to one up the surface provider?  

This is to make it more pronounced?  Should that be delegated to mIntens variable instead?

**Comment by Josharias** [comment]
msteiger,  I had a lovely time reviewing this.  So much more understanding about parts of this noise code from the way you have reorganized it.  Nice work.

File: modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinBaseSurfaceProvider.java (hunk 2 above)
Comment (Josharias) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinBaseSurfaceProvider.java]: Hrm,  I may be answering my own question.  msteiger,  maybe just confirm...

The previous implementation of the brownian noise was likely returning -2.11 to +2.11?

Comment (msteiger) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/PerlinBaseSurfaceProvider.java]: Yes, exactly! The original implementation had a range of -2.11 to +2.11 for 8 octaves and -2.12 to +2.12 for 9 octaves. The influence of additional layers (octaves) becomes exponentially smaller.

I think that the scale could be smaller (and less octaves) to get a less extreme environment, but world shaping was not part of this.

**Comment by msteiger** [comment]
Thanks and thanks for the review.

**Comment by Cervator** [comment]
Thanks for work, review and merge! :-)

**Pull request merged**